app/javascript/pages/ListingDetail.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useContext, useEffect } from "react"
```

Comment:
I'm wondering about what makes the best model for organizing files under the pages directory.

Right now we have:
- ListingDirectory
  - BuyHeader
  - DirectoryHelpers
  - GenericDirectory
  - etc
- BuyDirectory
- RentDirectory
- HomePage
- etc

Is that how you have it setup in bloom? with all top-level pages in pages/TopLevelPage.tsx, then directories with the same name that have related components and utility files?

Or would it make more sense to have:
- ListingDirectory
  - BuyDirectory
  - RentDirectory
  - BuyHeader
  - DirectoryHelpers
  - GenericDirectory
  - etc
- HomePage
- etc

cc @christine-sfg

Reply:
Updated the file structure based more closely on best practice, here is what it looks like now:

- components/ (reusable across pages)

- components/ (reusable across pages)
- modules/ (site sections like listings, applications, user management, all helpers and UI pieces live here)
- pages/ (only directly navigable pages, under directories that match their URL, lower case file names to differentiate from components)

Reply:
Yes! This is much better!